dbms/src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -525,6 +526,39 @@ ASTPtr MutationsInterpreter::prepareInterpreterSelectQuery(std::vector<Stage> &
         }
         select->setExpression(ASTSelectQuery::Expression::WHERE, std::move(where_expression));
     }
+    auto metadata = storage->getInMemoryMetadata();
+    /// We have to execute select in order of primary key
+    /// because we don't sort results additionaly and don't have
+    /// any guarantees on data order without ORDER BY. It's almost free, because we
+    /// have optimization for data read in primary key order.
+    if (metadata.order_by_ast)
+    {
+        ASTPtr dummy;
+
+        ASTPtr key_expr;
+        if (metadata.primary_key_ast)
+            key_expr = metadata.primary_key_ast;
+        else
+            key_expr = metadata.order_by_ast;
+
+        bool empty = false;
```

Comment:
What does this check for, something like `order by tuple()`? Can we just remove this check and pass the order by always, or does it break something? Btw I found that `order by tuple()` doesn't seem to work, because `tuple` requires at least one argument, but `order by 1` does.

Reply:
Yes we cannot order by `tuple()` and in this case we don't need order by at all.